packages/block-editor/src/components/block-list/style.native.scss

```diff
@@ -35,7 +35,6 @@
 }
 
 .containerStyleAddHere {
-	flex: 1;
```

Comment:
Removing the flex here avoided an issue where the "Add Block Here" text in the add block indicator would not display (or would appear and then disappear) when the related block content was not displayed. This fix seems to work very well, but I'm not 100% clear on what was causing the issue, so there may be better ways to address this.